frontend/app/(tabs)/favorites.tsx

```diff
@@ -1,10 +1,102 @@
-import { View, Text } from 'react-native';
-import React from 'react';
+import { View, Text, ScrollView, ActivityIndicator } from 'react-native';
+import { useState, useRef, useCallback } from 'react';
+import { Card } from '../components/Card';
+import { CardData } from '../types/card';
+import { fetchFavoriteCards, updateIdiom } from '../services/cardService';
+import { useTheme } from '../contexts/ThemeContext';
+import { useFocusEffect } from 'expo-router';
 
 const Favorites = () => {
+  const [favoriteCards, setFavoriteCards] = useState<CardData[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasLoaded, setHasLoaded] = useState(false);
+  const { colors } = useTheme();
+  const shouldRefresh = useRef(false);
+
+  const loadFavorites = useCallback(
+    async (forceRefresh = false) => {
+      if (!forceRefresh && hasLoaded) return;
+
+      setIsLoading(true);
+      try {
+        let allFavorites: CardData[] = [];
+        let page = 1;
+        let hasMore = true;
+        const limit = 50;
+
+        while (hasMore) {
+          const favoritesPage = await fetchFavoriteCards(page, limit);
+          allFavorites = allFavorites.concat(favoritesPage);
+
+          if (favoritesPage.length < limit) {
+            hasMore = false;
+          } else {
+            page += 1;
+          }
+        }
+
+        setFavoriteCards(allFavorites);
+        setHasLoaded(true);
+      } catch {
+        setFavoriteCards([]);
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    [hasLoaded],
+  );
+
+  useFocusEffect(
+    useCallback(() => {
+      if (!hasLoaded || shouldRefresh.current) {
+        loadFavorites(true);
+        shouldRefresh.current = false;
+      }
+    }, [hasLoaded, loadFavorites]),
+  );
+
+  const toggleFavorite = async (cardId: string) => {
```

Comment:
Don't we already have a similar method on the home component already? I think the favorites component should be virtually the same as the home component but we with this filtered state